applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -487,13 +539,13 @@ static void disconnect_subscriber(const void *subscriber_id)
 		return;
 	}
 
-	if (sub == state.active_subscriber) {
-		unlink_providers(sub);
+	if (sub->priority == state.active_subscriber_priority) {
+		uint8_t new_priority = get_active_sub_priority(sub);
 
-		state.active_subscriber = find_next_subscriber(sub);
-
-		if (state.active_subscriber) {
-			link_providers(state.active_subscriber);
+		if (new_priority != state.active_subscriber_priority) {
```

Comment:
What if we have two subscribers with the same priority? Shouldn't we link the other one that was not linked earlier?

Reply:
> Shouldn't the second subscriber be linked in that case? He is now the only connected subscriber with that priority.

A given provider can be linked to a subscriber only if the subscriber subscribes for a provider's HID input report (a subscriber may subscribe only for an arbitrarily selected subset of HID input reports - e.g. in USB selective HID reports subscriptions).